core/src/main/java/io/questdb/cairo/TxWriter.java

```diff
@@ -262,6 +266,38 @@ public void removeAttachedPartitions(long timestamp) {
         }
     }
 
+    public void removeActivePartition(
+            long newMinTimestamp,
+            long newMaxTimestamp,
+            long newFixedRowCount,
+            long newTransientRowCount,
+            long newColumnVersion
+    ) {
+        recordStructureVersion++;
+        prevMinTimestamp = minTimestamp;
+        minTimestamp = newMinTimestamp;
+        prevMaxTimestamp = maxTimestamp;
+        maxTimestamp = newMaxTimestamp;
+        fixedRowCount = newFixedRowCount;
+        prevTransientRowCount = transientRowCount;
+        transientRowCount = newTransientRowCount;
+        columnVersion = newColumnVersion;
+
+        partitionTableVersion++;
+        final int lim = attachedPartitions.size() - LONGS_PER_TX_ATTACHED_PARTITION;
+        if (lim > -1) {
+            attachedPartitions.setPos(lim);
+        }
+        txPartitionCount = getPartitionCount();
+        writeAreaSize = calculateWriteSize();
```

Comment:
Why you need to do all from this paragraph here? This is usually done in commit()

Reply:
indeed, to be fair I did say I needed to review the pr, this is done in commit.